cbcdb/main.py

```diff
@@ -0,0 +1,375 @@
+
+        Args:
+            sql: SQL string
+            params: List of parameters
+
+        Returns: None
+        """
+        if curs:
+            self._print_debug_output(f"Getting query:\n {sql}")
+            curs.execute(sql, params)
+            self._print_debug_output('csr.execute complete.')
+            conn.commit()
+            self._print_debug_output('conn.commit complete.')
+        else:
+            self._get_connection(sql, params, self.execute_simple)
+
+    def get_single_result(self, sql: str, params: list = None, curs=False, conn=False):
+        """
+        Execute as single SQL statement
+
+        Args:
+            sql: SQL string
+            params: List of parameters
+
+        Returns: None
+        """
+        if curs:
+            self._print_debug_output(f"Getting query:\n {sql}")
+            curs.execute(sql, params)
+            output = curs.fetchone()
+        else:
+            output = self._get_connection(sql, params, self.get_single_result)
+        return output
+
+    def execute_many(self, sql: str, params: list, curs=False, conn=False) -> None:
+        """
+        Executes a SQL Query
+
+        Args:
+            sql: SQL string
+            params: List of parameters
+
+        Returns: None
+        """
+        start_time = time.time()
+        self._print_debug_output(f"Execute Many: Inserting {len(params)} records")
+        self._print_debug_output(f"Getting query:\n {sql}")
+        params = self.convert_nan_to_none(params)
+        if curs:
+            execute_values(curs, sql, params)
+            duration = time.time() - start_time
+            self._print_debug_output(f'Inserted {len(params)} rows in {round(duration, 2)} seconds')
+            conn.commit()
+        else:
+            self._get_connection(sql, params, self.execute_many)
+
+    def delete(self, sql: str, params: list):
+        """
+        Deletes data
+
+        Args:
+            sql: SQL string
+            params: List of parameters
+
+        Returns: None
+        """
+        print('WARNING: Depreciated function. Use execute_simple.')
+        self.execute_simple(sql, params)
+
+    # Utility methods
+    @staticmethod
+    def make_column_names(columns: List[str]) -> str:
+        """
+        Creates the column name comma sep values
+        Args:
+            columns: A list of strings containing the column names
+
+        Returns: A comma separated string representing the columns for inclusion in a SQL statement.
+        """
+        return ','.join(columns)
+
+    @staticmethod
+    def make_param_string(input_list: List[str]) -> str:
+        """
+        Creates the %s for a query string
+        Args:
+            input_list: A list containing the parameters for a SQL call
+
+
+        Returns: A string with parameters represented by %s placeholders.
+        """
+        list_len = len(input_list)
+        param_string = ""
+        x = 0
+        while x < list_len:
```

Comment:
Minor point, but this could be replaced with a simple for loop over the elements of `input_list`

Reply:
Actually, we could simply create a list of `'%s'` strings of the same length as `input_list` (could do this with a list comprehension like: `placeholders = ['%s' for input in input_list]` and then call `', '.join(placeholders)` to create the param_string. This would remove the need for the slicing and the while loop